lib/std/os/uefi/tables/runtime_services.zig

```diff
@@ -23,50 +25,549 @@ pub const RuntimeServices = extern struct {
+        };
+
+        pointer = @bitCast(Bits{}),
+        optional = @bitCast(Bits{ .optional_ptr = true }),
+        _,
+    };
+
+    pub const PointerConversion = union(DebugDisposition) {
+        pointer: *anyopaque,
+        optional: ?*anyopaque,
+    };
+
+    pub const VariableAttributes = packed struct(u32) {
+        non_volatile: bool = false,
+        bootservice_access: bool = false,
+        runtime_access: bool = false,
+        hardware_error_record: bool = false,
+        /// Note: deprecated and should be considered reserved.
+        authenticated_write_access: bool = false,
+        time_based_authenticated_write_access: bool = false,
+        append_write: bool = false,
+        /// Indicates that the variable payload begins with a EFI_VARIABLE_AUTHENTICATION_3
+        /// structure, and potentially more structures as indicated by fields of
+        /// this structure.
+        enhanced_authenticated_access: bool = false,
+        _pad: u24 = 0,
+    };
+
+    pub const VariableAuthentication3 = extern struct {
+        version: u8 = 1,
+        type: Type,
+        metadata_size: u32,
+        flags: Flags,
+
+        pub fn payloadConst(self: *const VariableAuthentication3) []const u8 {
+            return @constCast(self).payload();
+        }
+
+        pub fn payload(self: *VariableAuthentication3) []u8 {
+            var ptr: [*]u8 = @ptrCast(self);
+            return ptr[@sizeOf(VariableAuthentication3)..self.metadata_size];
+        }
+
+        pub const Flags = packed struct(u32) {
+            update_cert: bool = false,
+            _pad: u31 = 0,
+        };
+
+        pub const Type = enum(u8) {
+            timestamp = 1,
+            nonce = 2,
+            _,
+        };
+    };
+
+    pub const VariableInfo = struct {
+        max_variable_storage_size: u64,
+        remaining_variable_storage_size: u64,
+        max_variable_size: u64,
+    };
+
+    pub const VariableNameIterator = struct {
+        pub const NextSizeError = uefi.UnexpectedError || error{
+            DeviceError,
+            Unsupported,
+        };
+
+        pub const IterateVariableNameError = NextSizeError || error{
+            BufferTooSmall,
+            NotFound,
+        };
+
+        services: *const RuntimeServices,
+        buffer: []u16,
+        guid: Guid,
+
+        pub fn nextSize(self: *VariableNameIterator) NextSizeError!?usize {
+            var len: usize = 0;
+            switch (self.services._getNextVariableName(
+                &len,
+                null,
+                &self.guid,
+            )) {
+                .success, .buffer_too_small => return len,
+                .not_found => return null,
+                .device_error => return Error.DeviceError,
+                .unsupported => return Error.Unsupported,
+                else => |status| return uefi.unexpectedStatus(status),
+            }
+        }
+
+        /// Call `nextSize` to ensure that `buffer` is large enough to hold the next
```

Comment:
This implies that `nextSize` will be responsible for resizing the buffer and it doesn't